src/puptoo/process/profile.py

```diff
@@ -644,6 +650,42 @@ def system_profile(
     if gcp_network_interfaces:
         profile["public_ipv4_addresses"] = _remove_empty_string(gcp_network_interfaces.public_ips)
 
+    if iris_cpfs and iris_list:
+        try:
+            iris_profile = {}
+            iris_profile["iris_system"] = True
+            iris_profile["iris_running_instances"] = []
+
+            # Get all CPF info in format <cpf-file-path: cpf-info>
+            iris_cpf_result = {}
+            for cpf in iris_cpfs:
+                if (cpf.file_path and cpf.has_option('ConfigFile', 'Product')
+                        and cpf.has_option('ConfigFile', 'Version')):
+                    iris_cpf_result[cpf.file_path] = {
+                        "product": cpf.get('ConfigFile', 'Product'),
+                        "version": cpf.get('ConfigFile', 'Version'),
+                    }
+            if iris_cpf_result:
+                # Filter for running instance and grab the instance_name
+                for instance in iris_list:
+                    if instance['status'].startswith('running'):
+                        instance_name = instance['instance_name']
+                        conf_directory = instance['directory']
+                        conf_file = instance['conf file'].split()[0].strip()
+                        cpf_file_path = os.path.join(conf_directory, conf_file)
+                        if instance_name and cpf_file_path and cpf_file_path in iris_cpf_result:
+                            instance_info = _remove_empties({
+                                "instance_name": instance['instance_name'],
+                                "product": iris_cpf_result[cpf_file_path]["product"],
+                                "version": iris_cpf_result[cpf_file_path]["version"],
+                            })
+                            if instance_info:
+                                iris_profile["iris_running_instances"].append(instance_info)
+            profile["iris"] = _remove_empties(iris_profile)
```

Comment:
I think we don't require `_remove_empties` here. Ultimately, we do apply this on the whole profile [here](https://github.com/RedHatInsights/insights-puptoo/blob/4700ed6de1e0287c6a4b228356f643d07d5d182c/src/puptoo/process/profile.py#L690). Let me know if this makes sense.

Reply:
Hey @r14chandra , 

Thanks for pointing this out too. 

I noticed that this `_remove_empties()` be called [here](https://github.com/RedHatInsights/insights-puptoo/blob/4700ed6de1e0287c6a4b228356f643d07d5d182c/src/puptoo/process/profile.py#L690) will only remove the empty values in first level of `profile`. 

And In this `profile["iris"]`, it may contains cases like: 
```
{
    "iris": {
        "iris_system": True,
        "iris_running_instances": []
    }
}
```
I am not sure if such empty `profile["iris"]["iris_running_instances"]` is acceptable to inventory service or not. Intent to remove the empty `profile["iris"]["iris_running_instances"]` here.

Any best practice for such corner cases?

Reply:
In that case, I think we should drop that field ie. `iris` altogether as this won't contain any relevant data. Does it make sense?

Reply:
> Any best practice for such corner cases?

It is hard to know if such an edge case as above is valid or not but I try a few cases that are possible and try to include some tests so that it would be easier for others to understand what they would be getting. 
Later if we get any issue with any scenario then we can anyway update the code and add a test around it.